### PR TITLE
Added shortcodes directory and added a shortcode for a discord invite.

### PR DESCRIPTION
**Pull request by ductapeofdoom**
Adds support for a discord invite shortcode.

Addresses #23.

Currently, the shortcode uses a static image from Discord's branding material as the base text, but that could be easily changed.

Additionally, as noted in the comments, the styling for the 'small', 'medium', and 'large' breakpoints is currently just width but can be styled as needed.

### CODE REVIEW

File: layouts/shortcodes/discordInvite.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<!--
```
Comment (nonowazu) [inline layouts/shortcodes/discordInvite.html:1]: I think we should replace this with a hugo comment:

```html
{{/* long
multiline comment here */}}
```

File: layouts/shortcodes/discordInvite.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Shortcode for Discord Invite
+Usage:
+< discordInvite invite_code=<> size=<> color=<> image=<> >
+ -->
+
+<!-- #5865F2 is discord's offical "blurple" color -->
+<!-- Large, medium, small breakpoints are arbitrary. Currently, just using tailwind width classes, but this can be tweaked however needed. -->
+<a {{ with .Get "invite_code" }} href="https://discord.gg/{{ . }}" {{ end }} >
+<img src="/img/Discord-Logo+Wordmark-White.png"
```
Comment (nonowazu) [inline layouts/shortcodes/discordInvite.html:10]: We should probably add this to the theme or pull that image from a cdn... does discord even have a cdn for this sort of thing?

File: layouts/shortcodes/discordInvite.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{{ if .Get "size" | eq "large"}}
```
Comment (nonowazu) [inline layouts/shortcodes/discordInvite.html:11]: I think you can do:

```html
{{ size := .Get "size" }}
```
And use that as a variable for the below parts.

**Review by nonowazu — CHANGES_REQUESTED** [review]
This is good work, thank you!

File: layouts/shortcodes/discordInvite.html (hunk 2 above)
Comment (ductapeofdoom) [reply layouts/shortcodes/discordInvite.html:10]: I tried just linking to the image on https://discord.com/branding but it appear Discord blocks linking to that that so I think we would have to self host to use it. I mainly used to for filler so whatever we need to change it to is fine.

File: layouts/shortcodes/discordInvite.html (hunk 1 above)
Comment (ductapeofdoom) [reply layouts/shortcodes/discordInvite.html]: Updated in latest commit. Still learning hugo syntax so thanks for the recommendation.

File: layouts/shortcodes/discordInvite.html (hunk 3 above)
Comment (ductapeofdoom) [reply layouts/shortcodes/discordInvite.html]: Updated in latest commit.

**Review by nonowazu — APPROVED** [review]
Thank you for getting this in!

**Pull request merged**